### PR TITLE
Move to core-api UPayload data model

### PR DESCRIPTION
**Pull request by stevenhartley**
Pulling from snapshot till the other project is merged

#40

### CODE REVIEW

**Review by neelam-kushwah — APPROVED** [review]

File: src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java [hunk 1]
```diff
@@ -25,14 +25,20 @@
...
+import com.google.protobuf.ByteString;
 import com.google.protobuf.Int32Value;
 import com.google.protobuf.InvalidProtocolBufferException;
 import com.google.rpc.Code;
 import com.google.rpc.Status;
 import org.eclipse.uprotocol.transport.builder.UAttributesBuilder;
-import org.eclipse.uprotocol.transport.datamodel.UPayload;
 import org.eclipse.uprotocol.uri.serializer.LongUriSerializer;
-import org.eclipse.uprotocol.v1.*;
+import org.eclipse.uprotocol.v1.UUri;
```
Comment (tamarafischer) [inline src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java:35]: why do we have the version in the package name. makes upgrading even messier

File: src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java [hunk 2]
```diff
@@ -64,8 +73,10 @@ public CompletableFuture<UPayload> invokeMethod(UUri topic, UPayload payload, UA
         public CompletableFuture<UPayload> invokeMethod(UUri topic, UPayload payload, UAttributes attributes) {
             Status status = Status.newBuilder().setCode(Code.INVALID_ARGUMENT_VALUE).setMessage("boom").build();
             Any any = Any.pack(status);
-            UPayload data = new UPayload(any.toByteArray(), UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF);
-
+            UPayload data = UPayload.newBuilder()
```
Comment (tamarafischer) [inline src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java:76]: personally I prefer working with the constructors and not verbose builders if they are not really needed. 
I do understand why you wanted to define the payload as a protobuf object
But then again - at the end of the day, uTransport gets a byte[], so are we not creating a protobuf object whose job is transport and then just translating it into the **real** transport object which is the byte[]
what value is achieved by the payload being a protobuf object?

**Review by tamarafischer — APPROVED** [review]

File: src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java (hunk 1 above)
Comment (stevenhartley) [reply src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java:35]: This is to ensure that if/when we go to a non-backwards compatible version of uProtocol (ex. v1) that we have differentiate the data model between them like we do for uSubscription, uDiscovery, etc...

File: src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java (hunk 2 above)
Comment (stevenhartley) [reply src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java:76]: UPayload is to wrap the data + metadata and keep it separate from the message attributes (priority, addressing, timeout, etc...). Yes you are right for the data but the metadata will most likely be mapped somehow into the local transport (HTTP header, zenoh header, etc...). 
the benefit of the UPayload being in proto is the consistently for all data types (and the UPayloadFormat as well).

**Pull request merged**